src/dune_async_io/async_io.ml

```diff
@@ -201,6 +206,9 @@ let rec select_loop t =
      | exception Unix.Unix_error (Unix.(EINTR | EAGAIN), _, _) ->
        Mutex.lock t.mutex;
        select_loop t
+     | exception exn ->
+       let exn = Exn_with_backtrace.capture exn in
+       raise (Unlocked exn)
```

Comment:
Should this be `raise_notrace`?

Reply:
Sure, why not. Doesn't make much of a difference.